internal/controllers/topology/cluster/reconcile_state_test.go

```diff
@@ -2042,6 +2195,341 @@ func TestReconcileMachineDeployments(t *testing.T) {
+	bootstrapConfig5 := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-5").Build()
+	mp5 := newFakeMachinePoolTopologyState("mp-5", infrastructureMachinePool5, bootstrapConfig5)
+	infrastructureMachinePool5WithChangedKind := infrastructureMachinePool5.DeepCopy()
+	infrastructureMachinePool5WithChangedKind.SetKind("ChangedKind")
+	mp5WithChangedinfrastructureMachinePoolKind := newFakeMachinePoolTopologyState("mp-4", infrastructureMachinePool5WithChangedKind, bootstrapConfig5)
+
+	infrastructureMachinePool6 := builder.TestInfrastructureMachinePool(metav1.NamespaceDefault, "infrastructure-machinepool-6").Build()
+	bootstrapConfig6 := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-6").Build()
+	mp6 := newFakeMachinePoolTopologyState("mp-6", infrastructureMachinePool6, bootstrapConfig6)
+	bootstrapConfig6WithChangedNamespace := bootstrapConfig6.DeepCopy()
+	bootstrapConfig6WithChangedNamespace.SetNamespace("ChangedNamespace")
+	mp6WithChangedbootstrapConfigNamespace := newFakeMachinePoolTopologyState("mp-6", infrastructureMachinePool6, bootstrapConfig6WithChangedNamespace)
+
+	infrastructureMachinePool7 := builder.TestInfrastructureMachinePool(metav1.NamespaceDefault, "infrastructure-machinepool-7").Build()
+	bootstrapConfig7 := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-7").Build()
+	mp7 := newFakeMachinePoolTopologyState("mp-7", infrastructureMachinePool7, bootstrapConfig7)
+
+	infrastructureMachinePool8Create := builder.TestInfrastructureMachinePool(metav1.NamespaceDefault, "infrastructure-machinepool-8-create").Build()
+	bootstrapConfig8Create := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-8-create").Build()
+	mp8Create := newFakeMachinePoolTopologyState("mp-8-create", infrastructureMachinePool8Create, bootstrapConfig8Create)
+	infrastructureMachinePool8Delete := builder.TestInfrastructureMachinePool(metav1.NamespaceDefault, "infrastructure-machinepool-8-delete").Build()
+	bootstrapConfig8Delete := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-8-delete").Build()
+	mp8Delete := newFakeMachinePoolTopologyState("mp-8-delete", infrastructureMachinePool8Delete, bootstrapConfig8Delete)
+	infrastructureMachinePool8Update := builder.TestInfrastructureMachinePool(metav1.NamespaceDefault, "infrastructure-machinepool-8-update").Build()
+	bootstrapConfig8Update := builder.TestBootstrapConfig(metav1.NamespaceDefault, "bootstrap-config-8-update").Build()
+	mp8Update := newFakeMachinePoolTopologyState("mp-8-update", infrastructureMachinePool8Update, bootstrapConfig8Update)
+	infrastructureMachinePool8UpdateWithChanges := infrastructureMachinePool8Update.DeepCopy()
+	infrastructureMachinePool8UpdateWithChanges.SetLabels(map[string]string{"foo": "bar"})
+	bootstrapConfig8UpdateWithChanges := bootstrapConfig8Update.DeepCopy()
+	bootstrapConfig8UpdateWithChanges.SetLabels(map[string]string{"foo": "bar"})
```

Comment:
I think we should set something in spec instead to simulate changes (also for other places)

Reply:
Since the name is not updated for MachinePools like when Machines are updated, would it be sufficient to check for equivalence using `GetResourceVersion` like so: https://github.com/kubernetes-sigs/cluster-api/pull/9380/files?diff=unified&w=0#diff-a0cab4d80f6281821063eccb1ab5eff3712ff3afd257c0c2e94c4b965f324bd1R2499